internal/pkg/skuba/kubernetes/node_version_test.go

```diff
@@ -18,394 +18,1027 @@
+			clusterVersion:   clusterVersion,
+			expectReturn:     true,
+		},
+		{
+			name:             "master node (kubelet) minor version less than cluster version",
+			isMaster:         true,
+			apiServerVersion: clusterVersion,
+			kubeletVersion:   minorLess,
+			clusterVersion:   clusterVersion,
+			expectReturn:     true,
+		},
+		{
+			name:           "worker node same as cluster version",
+			kubeletVersion: clusterVersion,
+			clusterVersion: clusterVersion,
+			expectReturn:   false,
+		},
+		{
+			name:           "worker node major version less than cluster version",
+			kubeletVersion: majorLess,
+			clusterVersion: clusterVersion,
+			expectReturn:   true,
+		},
+		{
+			name:           "worker node minor version less than cluster version",
+			kubeletVersion: minorLess,
+			clusterVersion: clusterVersion,
+			expectReturn:   true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			var nodeVersionInfo NodeVersionInfo
+			switch tt.isMaster {
+			case true:
+				nodeVersionInfo = NodeVersionInfo{
+					APIServerVersion: &tt.apiServerVersion,
+					KubeletVersion:   &tt.kubeletVersion,
+				}
+			default:
+				nodeVersionInfo = NodeVersionInfo{
+					KubeletVersion: &tt.kubeletVersion,
+				}
+			}
+			actualReturn := nodeVersionInfo.DriftsFromClusterVersion(&tt.clusterVersion)
+			if actualReturn != tt.expectReturn {
+				t.Errorf("got: (%v), want: (%v)", actualReturn, tt.expectReturn)
+			}
+		})
+	}
+}
+
+func TestToleratesFromClusterVersion(t *testing.T) {
+	clusterVersion := *version.MustParseSemantic("v1.1.1")
+	majorLess := *version.MustParseSemantic("v0.1.1")
+	majorMore := *version.MustParseSemantic("v2.1.1")
+	minorLess := *version.MustParseSemantic("v1.0.1")
+	minorMore := *version.MustParseSemantic("v1.2.1")
+	tests := []struct {
+		name             string
+		isMaster         bool
+		apiServerVersion version.Version
+		kubeletVersion   version.Version
+		expectReturn     bool
+	}{
+		{
+			name:             "master node version same as cluster version",
+			isMaster:         true,
+			apiServerVersion: clusterVersion,
+			kubeletVersion:   clusterVersion,
+			expectReturn:     true,
+		},
+		{
+			name:             "master node (apiserver) minor version is one less than cluster version",
+			isMaster:         true,
+			apiServerVersion: minorLess,
+			expectReturn:     false,
+		},
+		{
+			name:             "master node (apiserver) minor version is one greater than cluster version",
+			isMaster:         true,
+			apiServerVersion: minorMore,
+			expectReturn:     false,
+		},
+		{
+			name:             "master node (apiserver) major version is one less than cluster version",
+			isMaster:         true,
+			apiServerVersion: majorLess,
+			expectReturn:     false,
+		},
+		{
+			name:             "master node (apiserver) major version is one greater than cluster version",
+			isMaster:         true,
+			apiServerVersion: majorMore,
+			expectReturn:     false,
+		},
```

Comment:
[1]

Reply:
This case expects to return before check for kubeletVersion.
```
if nvi.IsControlPlane() {
		if nvi.APIServerVersion.Major() != clusterVersion.Major() ||
			nvi.APIServerVersion.Minor() != clusterVersion.Minor() {
			return false
		}
	}
```